src/lib/mongoBackend/mongoQueryContext.cpp

```diff
@@ -195,10 +203,13 @@ void addContextProviders(ContextElementResponseVector& cerV, ContextRegistration
 
     if (cr.contextRegistrationAttributeVector.size() == 0)
     {
-      /* Registration without attributes */
-      for (unsigned int jx = 0; jx < cr.entityIdVector.size(); ++jx)
+      if (!limitReached)
       {
-        addContextProviderEntity(cerV, cr.entityIdVector.get(jx), cr.providingApplication);
+        /* Registration without attributes */
+        for (unsigned int jx = 0; jx < cr.entityIdVector.size(); ++jx)
+        {
+          addContextProviderEntity(cerV, cr.entityIdVector.get(jx), cr.providingApplication);
```

Comment:
Regarding [  operator, we have already an issue about this (https://github.com/telefonicaid/fiware-orion/issues/810) so I think it would be more efficient to solve all them in a shot when that issue get addressed.

Regardingl jx -> elx, I will implement the change.